lib/s3.py

```diff
@@ -40,6 +50,37 @@ def is_key_present(url):
     return get_key(url.hostname) is not None
 
 
+def sign_request(url, method='GET', checksum=SHA256_NIL, headers={}, region='any', service='S3'):
```

Comment:
This could do with a docstring. Particularly, what does the checksum mean (it's the checksum of what exactly?) Does it really make sense to default to NIL here, instead of making it explicit in the caller what's going on?

What other "service" values are supported? Given that this is an "S3" library, perhaps we should start with not  having this parameter until we actually need it?

Plus, document the return value.